src/com/xilinx/rapidwright/examples/ParamCounterModule.java

```diff
@@ -0,0 +1,257 @@
+            accepts(SLICE_SITES_OPT).withOptionalArg().defaultsTo(sliceSite).describedAs("Lower left slice to be used for adder/subtracter");
+            accepts(COUNT_DOWN_OPT).withOptionalArg().ofType(Boolean.class).defaultsTo(countDown).describedAs("Counts down instead of up");
+            accepts(STEP_OPT).withOptionalArg().ofType(Long.class).defaultsTo(step).describedAs("Counts down instead of up");
+            accepts(INIT_OPT).withOptionalArg().ofType(Long.class).defaultsTo(initValue).describedAs("Counts down instead of up");
+            accepts(VERBOSE_OPT).withOptionalArg().ofType(Boolean.class).defaultsTo(verbose).describedAs("Print verbose output");
+            acceptsAll( Arrays.asList(HELP_OPT, "?"), "Print Help" ).forHelp();
+        }};
+
+        return p;
+    }
+
+    private static void printHelp(OptionParser p) {
+        MessageGenerator.printHeader("Adder/Subtractor Generator");
+        System.out.println("This RapidWright program creates a placed and routed DCP for UltraScale+ devices that contains \n"
+                + "imported into UltraScale+ designs to perform addition or subtraction.  See \n"
+                + "RapidWright documentation for more information.\n");
+        try {
+            p.accepts(OUT_DCP_OPT).withOptionalArg().defaultsTo("slr_crosser.dcp").describedAs("Output DCP File Name");
+            p.printHelpOn(System.out);
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void main(String[] args) {
+        // Extract program options
+        OptionParser p = createOptionParser();
+        OptionSet opts = p.parse(args);
+        boolean verbose = (boolean) opts.valueOf(VERBOSE_OPT);
+        if (opts.has(HELP_OPT)) {
+            printHelp(p);
+            return;
+        }
+
+        CodePerfTracker t = verbose ? new CodePerfTracker(AddSubGenerator.class.getSimpleName(),true).start("Init") : null;
+
+        String partName = (String) opts.valueOf(PART_OPT);
+        String designName = (String) opts.valueOf(DESIGN_NAME_OPT);
+        String adderName = (String) opts.valueOf(ADDER_NAME_OPT);
+        String outputDCPFileName = (String) opts.valueOf(OUT_DCP_OPT);
+        String clkName = (String) opts.valueOf(CLK_NAME_OPT);
+        double clkPeriodConstraint = (double) opts.valueOf(CLK_CONSTRAINT_OPT);
+        int width = (int) opts.valueOf(WIDTH_OPT);
+        String sliceName = (String) opts.valueOf(SLICE_SITES_OPT);
+        boolean countDown = (boolean) opts.valueOf(COUNT_DOWN_OPT);
+        long step = (long) opts.valueOf(STEP_OPT);
+        long initValue = (long) opts.valueOf(INIT_OPT);
+
+
+        // Perform some error checking on inputs
+        Part part = PartNameTools.getPart(partName);
+        if (part == null || part.isSeries7()) {
+            throw new RuntimeException("ERROR: Invalid/unsupport part " + partName + ".");
+        }
+
+        if (verbose) t.stop().start("Create Add/Sub Module");
```

Comment:
Could we move all the counter related functionality into its own method?  Ideally, we would like to have a reusable method like `createCounter()` (similar to `createAddSub()`)?